### PR TITLE
Doc rule editor usage

### PR DESCRIPTION
**Pull request by kylebrandt**

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -159,9 +159,70 @@ The open incident filter supports joining terms in `()` as well as the `AND`, `O
...
+
+## Rule Editor Image
+![Rule Editor Image](/public/rule_editor.jpg)
+
+## Textarea
+The text area will be loaded with the running config when the Rule Editor view is loaded. A hash of the config when you start editing it is saved. If someone else edits the UI and saves it, Bosun will detect that the config hash has changed and show a warning above the text area.
+
+When you run test your version of the config is saved in Bosun, and you can link to it so others can see it.
+
+The editor is build using the open source [Ace editor](https://ace.c9.io/).
```
Comment (dimamedvedev) [inline docs/usage.md]: built

File: docs/usage.md [hunk 2]
```diff
@@ -159,9 +159,70 @@ The open incident filter supports joining terms in `()` as well as the `AND`, `O
...
+
+An external command to run on saves can also be defined with the [CommandHookPath setting in the system configuration](/system_configuration#commandhookpath). This can be used to do things like create backups of the file or check the changes into version control. If this command returns a non-zero exit code, saving will also fail.
+
+In all cases where a save fails, a reload will not happen and the save will not be persisted (the definitions file will not be changed).
+
+## Alert Testing
+Alerts can be tested before they are committed to production. This allows you to refine the trigger conditions to control the signal to noise and to preview the rendered templates to make sure alerts are informative. This done by selecting the alert the from the [Jump Alert Drop down](http://localhost:4000/usage#jump-buttons) at <a href="/usage#rule-editor-image" class="image-number">①</a> and the clicking the test alert button at <a href="/usage#rule-editor-image" class="image-number">④</a>.
```
Comment (dimamedvedev) [inline docs/usage.md]: wrong link, "localhost:4000"

File: docs/usage.md [hunk 3]
```diff
@@ -159,9 +159,70 @@ The open incident filter supports joining terms in `()` as well as the `AND`, `O
...
+
+Which item from the result set that will be rendered in the Template tab is controlled by the <span class="docFromLabel">Template Group</span> field at <a href="/usage#rule-editor-image" class="image-number">④</a>. Which result to use for the template is picked by specifying a tagset in the format of `key=value,key=value`. The first result that has the specified tags will be used. If no results match, than the first result is chosen.
+
+<div class="admonition">
+<p class="admonition-title">Tip</p>
+<p>When working on a template it is good to set the <span class="docFromLabel">From</span> time to a fixed date. That way when expressions are rerun they will likely hit Bosun's query cache and things will be faster.</p>
+</div>
+
+The <span class="docFromLabel">Email</span> field at <a href="/usage#rule-editor-image" class="image-number">④</a> makes it so when an alert is tested, the rendered template is email to the address specified in the field. This is so you can check for any differences between what you see in the <span class="docFromLabel">Template</span> tab.
```
Comment (dimamedvedev) [inline docs/usage.md]: "is emailed"?

File: docs/usage.md [hunk 4]
```diff
@@ -159,9 +159,70 @@ The open incident filter supports joining terms in `()` as well as the `AND`, `O
...
+## Jump Buttons
+The Jump drop downs <a href="/usage#rule-editor-image" class="image-number">①</a> will take you to defined sections within the config. In particular, the alert drop down selects which alert will be used for testing.
+
+At the end there is a switcher that can be used when you are working on an alert. It allows you to just back and forth between the alert and the alert referenced in the template.
+
+## Download / Validate
+The download button <a href="/usage#rule-editor-image" class="image-number">②</a> will download the config file as a text file. Validate makes sure that Bosun considers the config valid using the same validation that is required for Bosun to start.
+
+## Definition [Rule] Saving
+The save button <a href="/usage#rule-editor-image" class="image-number">②</a> will bring up a dialogue that lets you save the config. This only appears if you have permission to save the config, and tbe [system configuration's `EnableSave`](/system_configuration#enablesave) has been set to true.
```
Comment (cleonte) [inline docs/usage.md]: the [system configuration's `EnableSave`](/system_configuration#enablesave)

**Pull request merged**